### ISSUE TITLE
Ensure extrinsics in the XCM pallet check for a configurable origin

### ISSUE THREAD

**Issue by KiChjang** [paritytech/polkadot#6442]
Code patterns like the following:

```rust
pub fn force_unsubscribe_version_notify(
	origin: OriginFor<T>,
	location: Box<VersionedMultiLocation>,
) -> DispatchResult {
	ensure_root(origin)?;
	// ...
}
```

are common in the XCM pallet. This is undesirable as only root can call these extrinsics. Instead, going forward, what we should do is create a new config item called `AdminOrigin`, and check for that. This will allow the origin to have its own curve in Gov2.

**Comment by serkul** [comment]
Hi @KiChjang, can you take a look at the PR which I made for this issue? 
I see that CI fails in cumulus, because `AdminOrigin` is missing there in `impl pallet_xcm...`. I don't know how to address this correctly since it's a separate repo. Maybe I misunderstood the task, but, how I see it, adding a new item in a pallet's config will require changes in all places where the pallet's config is implemented.

**Comment by KiChjang** [comment]
This is not meant to be worked on just yet; it's an issued filed specifically for XCM v3.

**Comment by serkul** [comment]
Ok, it wasn't obvious for me. I just saw an "easy" tag and thought it was a good issue to start with :) I'll close the PR then.

**Comment by KiChjang** [comment]
Since XCM v3 has just merged, this is free to be worked upon again.

**Comment by serkul** [comment]
@KiChjang, I see this issue stopped being "easy". Can I continue to work on it or should I leave it alone?

**Comment by KiChjang** [comment]
It's simply because we've removed that label, so I can't apply it on this issue. You are welcome to work on it.

**Comment by serkul** [comment]
after rebasing, there are `Currency`, `CurrencyMatcher`, `TrustedLockers`, `SovereignAccountOf`, `MaxLockers`, `WeightInfo` types that are missing in pallet_xcm::Config in runtime/rococo. Does it need to be fixed or I just reopen PR with some tests failing?

**Comment by KiChjang** [comment]
I unfortunately do not have the context of what you're talking about. It is likely to be a rebasing issue.

**Comment by serkul** [comment]
You are right, Keith , it was a rebasing issue. I opened a new [PR] (https://github.com/paritytech/polkadot/pull/6632). There are two label checks which fail, I guess, I cannot fix them myself.
Another failing check,  [as before XCM v3 merge](https://github.com/paritytech/polkadot/issues/6442#issuecomment-1365885206) ", is continuous-integration/gitlab-check-dependent-cumulus", because of the new config item.

**Comment by KiChjang** [comment]
Fixed in #6928.
